kotlinx-coroutines-core/common/src/channels/AbstractChannel.kt

```diff
@@ -633,10 +634,10 @@ internal abstract class AbstractChannel<E> : AbstractSendChannel<E>(), Channel<E
         @JvmField var resumeToken: Any? = null
         @JvmField var pollResult: E? = null
 
-        override fun failure(affected: LockFreeLinkedListNode, next: Any): Any? {
-            if (affected is Closed<*>) return affected
-            if (affected !is Send) return POLL_FAILED
-            return null
+        override fun failure(affected: LockFreeLinkedListNode, next: Any): Any? = when (affected) {
```

Comment:
It is possible to get rid of the copy-paste here as well, for example:
```
override fun failure(affected: LockFreeLinkedListNode, next: Any): Any? = enqueueFailure(affected) { POLL_FAILED.takeIf { it !is Send } }
```

Reply:
Current code is easier to understand. Postone extraction of common code for future optimization.